src/cmd/create.go

```diff
@@ -710,19 +710,50 @@ func pullImage(image, release, authFile string) (bool, error) {
 		shouldPullImage = true
 	}
 
+	imageSizeChan := make(chan string)
+	confirmationChan := make(chan bool)
+	var userResponded bool
+
 	if promptForDownload {
 		fmt.Println("Image required to create toolbox container.")
 
-		var prompt string
+		prompt := fmt.Sprintf("Download %s (...)? [y/N]:", imageFull)
+		go func() {
+			shouldPullImage = askForConfirmation(prompt)
+			confirmationChan <- shouldPullImage
+		}()
 
-		if imageSize, err := getImageSizeFromRegistry(imageFull); err != nil {
-			logrus.Debugf("Getting image size failed: %s", err)
-			prompt = fmt.Sprintf("Download %s? [y/N]:", imageFull)
-		} else {
-			prompt = fmt.Sprintf("Download %s (%s)? [y/N]:", imageFull, imageSize)
+		// Launch a goroutine to call getImageSizeFromRegistry and send the size to the channel
+		go func() {
+			imageSize, err := getImageSizeFromRegistry(imageFull)
+			if err != nil {
+				logrus.Debugf("Getting image size failed: %s", err)
+				imageSize = ""
+			}
+			imageSizeChan <- imageSize
+		}()
+
+		var imageSize string
+		select {
+		case imageSize = <-imageSizeChan:
+			if imageSize != "" {
+				prompt = fmt.Sprintf("\rDownload %s (%s)? [y/N]:", imageFull, imageSize)
+			} else {
+				prompt = fmt.Sprintf("\rDownload %s? [y/N]:", imageFull)
+			}
+			fmt.Printf("%s", prompt)
```

Comment:
If we have an `askForConfirmationAsync` then the second updated prompt will be displayed by it, and it won't lose the image size.